### ISSUE TITLE
right click context menu unexpectedly closes itself.

### ISSUE THREAD

**Issue by Turmfalke2** [ValveSoftware/steam-for-linux#9684]
Hi,
In the right click context menus of the new steam, if you move your mouse cursor from a nested menu entry back to a normal one the menu flickers out of existence.

If I move the cursor from **View Profile**
![screen04](https://github.com/ValveSoftware/steam-for-linux/assets/5174562/f587f958-2d9a-43d4-80de-b4b42dda614c)
to **Trading**
![screen05](https://github.com/ValveSoftware/steam-for-linux/assets/5174562/42fce170-fb61-4fb8-b060-6176e51c0c99)
and back again the menu disappears.

This is unexpected and inconsistent with right click menus in other programs like my browser here.

#### Your system information

* Steam client version (build number or date): 1686880776
* Distribution (e.g. Ubuntu): Gentoo
* Opted into Steam client beta?: No
* Have you checked for system updates?: Yes
* Steam Logs: [generate by running this command in a terminal `tar -zcvf ~/Desktop/steam-logs.tar.gz ~/.steam/steam/logs`] 
[steam-logs.tar.gz](https://github.com/ValveSoftware/steam-for-linux/files/11805533/steam-logs.tar.gz)

* GPU: Nvidia

**Comment by kisak-valve** [comment]
Hello @Turmfalke2, please copy your system information from Steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

**Comment by Turmfalke2** [comment]
Here: https://gist.github.com/Turmfalke2/09809db3b1e4c1afe861af6d7a416f49
It looks smaller than the one from the old client.

The steamwebhelper.log is getting spammed with errors like this:
```
X Error of failed request:  BadAlloc (insufficient resources for operation)
Major opcode of failed request:  151
Serial number of failed request:  4139
Webhelper: An X Error occurred
X Error of failed request:  BadDrawable (invalid Pixmap or Window parameter)
Major opcode of failed request:  155
Resource id in failed request:  0x2600658
Serial number of failed request:  4140
Webhelper: An X Error occurred
X Error of failed request:  GLXBadPixmap
Major opcode of failed request:  151
Serial number of failed request:  0
Webhelper: An X Error occurred
X Error of failed request:  GLXBadPixmap
Major opcode of failed request:  151
Serial number of failed request:  0
Webhelper: An X Error occurred
```
but they apparently aren't deadly? Doesn't look great.

**Comment by Turmfalke2** [comment]
focus_follows_mouse is false by default, but toggling it to true makes no difference. I don't think the context menu counts as an extra window for the x server.

**Comment by kisak-valve** [comment]
Could be herbstluftwm specific behavior. It might be interesting to test how Steam behaves with a different window manager.

**Comment by Turmfalke2** [comment]
Same behaviour when using xfce4

**Comment by chaoticsoap** [comment]
Same happening on my end, KDE Plasma, Arch linux

`
Computer Information:
Manufacturer: Micro-Star International Co., Ltd.
Model: B450M PRO-VDH MAX (MS-7A38)
Form Factor: Desktop
No Touch Input Detected
Processor Information:
CPU Vendor: AuthenticAMD
CPU Brand: AMD Ryzen 7 5800X3D 8-Core Processor
CPU Family: 0x19
CPU Model: 0x21
CPU Stepping: 0x2
CPU Type: 0x0
Speed: 4548 MHz
16 logical processors
8 physical processors
Hyper-threading: Supported
FCMOV: Supported
SSE2: Supported
SSE3: Supported
SSSE3: Supported
SSE4a: Supported
SSE41: Supported
SSE42: Supported
AES: Supported
AVX: Supported
AVX2: Supported
AVX512F: Unsupported
AVX512PF: Unsupported
AVX512ER: Unsupported
AVX512CD: Unsupported
AVX512VNNI: Unsupported
SHA: Supported
CMPXCHG16B: Supported
LAHF/SAHF: Supported
PrefetchW: Unsupported
Operating System Version:
"Arch Linux" (64 bit)
Kernel Name: Linux
Kernel Version: 6.3.8-zen1-1-zen
X Server Vendor: The X.Org Foundation
X Server Release: 12101008
X Window Manager: KWin
Steam Runtime Version: steam-runtime_0.20230509.49499
Video Card:
Driver: AMD AMD Radeon RX 6700 XT (navi22, LLVM 15.0.7, DRM 3.52, 6.3.8-zen1-1-zen)
Driver Version: 4.6 (Compatibility Profile) Mesa 23.1.2
OpenGL Version: 4.6
Desktop Color Depth: 24 bits per pixel
Monitor Refresh Rate: 165 Hz
VendorID: 0x1002
DeviceID: 0x73df
Revision Not Detected
Number of Monitors: 2
Number of Logical Video Cards: 1
Primary Display Resolution: 1920 x 1080
Desktop Resolution: 3840 x 1168
Primary Display Size: 20.75" x 11.65" (23.78" diag), 52.7cm x 29.6cm (60.4cm diag)
Primary VRAM: 12288 MB
Sound card:
Audio device: ATI R6xx HDMI
Memory:
RAM: 32017 Mb
VR Hardware:
VR Headset: None detected
Miscellaneous:
UI Language: English
LANG: en_US.UTF-8
Total Hard Disk Space Available: 223753 MB
Largest Free Hard Disk Block: 151644 MB
Storage:
Number of SSDs: 2
SSD sizes: 1000G,240G
Number of HDDs: 0
Number of removable drives: 0
`

**Comment by Turmfalke2** [comment]
@kisak-valve I can also reproduce it on the steam deck when opening steam in desktop mode.

That should make it a bit easier to reproduce for you.

**Comment by eslowney** [comment]
Same issue.  I haven't opened Steam in a month or two, so I just upgraded, and the GUI looks a little different, and a right-click on my Friend's list flickers out, while in the upper left corner of my monitor a window blinks in and out of existence too quickly to see.

I'm using Xubuntu 22.04, so Ubuntu with XFCE.

**Comment by apocalyptech** [comment]
I hadn't noticed this bug when searching, so I'd created a duplicate of my own.  Some notes from my duplicate which haven't been mentioned here, from that bug:

I suspect it might have something to do with the glitchy UIs described in https://github.com/ValveSoftware/steam-for-linux/issues/9614 -- in the second of the videos I posted there, you can see glitchiness out on the lefthand side of the screen when the sub-menus are collapsed, which almost looks like Steam is creating a new window over there. Perhaps that new window is temporarily stealing focus, which makes the context menu lose focus, which triggers the context-close. Here's that video for reference here, too (you may need to watch in 60fps to actually see the glitching): https://www.youtube.com/watch?v=CBA2PBmAq2Y

It's worth noting that this only appears to happen in Small Mode -- when in Large Mode I can get to those menus just fine (though the extremely unpleasant visual glitching is omnipresent no matter what the mode).

**Comment by Turmfalke2** [comment]
Appears to be fixed in the steam beta.

**Comment by Zorrototo** [comment]
This is the same exact behavior on the current BETA for me.

**Comment by Turmfalke2** [comment]
Yeah sorry, my bad. You are right. I did my retest with the steam beta using the library menu, the right click context there works.

It's still broken in the friends list.

**Comment by nPHYN1T3** [comment]
I am seeing this occasionally and with the Context Menu Compatibility Mode Enabled it's STILL clicking things UNDER the menu's when they don't just vanish on their own.

**Comment by zealws** [comment]
Same issue on Xorg/Cinnamon/Arch Linux.
Here's [the system information from steam](https://gist.github.com/zealws/f039e2abc5dd305c500647a11dfb67f7) for my setup.

I upgraded to the steam client beta but the issue was still present in the beta so I went back to the stable client (`Steam Version:  1690583737`).

In the stable version of the client, enabling `Steam > Settings > Interface > Enable context menu focus compatibility mode` seemed to help with immediate context menus (e.g. Right click a game, and click "Properties" works now but didn't without that setting enabled) but not for nested menus (e.g. Right click a game, mouse over "Manage" click "Browse local files" does not work). This behavior seemed the same in the current beta version.

On the stable client, changing `Window focus mode` from `Mouse` to `Click` in cinnamon-settings-manager resolved the issue, although I think steam should play nicely with mouse-based focus as I primarily use that. I did not try this on the beta client to know if there's a change in behavior.

**Comment by pirat89** [comment]
I see the same behaviour as @zealws  described. Enabling of `context menu focus compatibility`  partially resolved the problem. 
* Fedora 38,
* X.org,
* steam version: 1696019606

**Comment by Root-Core** [comment]
@Zorrototo has posted a nice video in issue #9937 - https://www.youtube.com/watch?v=JTKE3SvoqJM

---------------

This is another annoying issue. If you click fast enough, you might manage to join a game.
It is a bit embarrassing if you tell your friends to use Linux and you can not even join their game. :/

`context menu focus compatibility` is not helping at all.

- Manjaro 23.1.1
- Linux 6.6.8-2-MANJARO
- KDE 5.27.10
- Wayland

**Comment by red15** [comment]
Having the same, not only in beta.

https://gist.github.com/red15/02c7ab18b587c9be4ce1674e2e4759fe

**Comment by Zorrototo** [comment]
I noticed a new thing now on Steam BETA, right click on someone in your friend list, and then hover the menu items back and forth, you'll see some kind of window popping in the top left corner of monitor when the context menu just doesn't crash completely : https://youtu.be/lyvVhosAJ7A

**Comment by RexSonic** [comment]
> I noticed a new thing now on Steam BETA, right click on someone in your friend list, and then hover the menu items back and forth, you'll see some kind of window popping in the top left corner of monitor when the context menu just doesn't crash completely : https://youtu.be/lyvVhosAJ7A

I am having the same issue on arch with kde plasma 5

**Comment by Conzar** [comment]
I am also having this issue.  I tried upgrading to Beta and the issue still remains.

**Comment by mstenseth** [comment]
how is this a problem , just use a web ui and give your junior devs a vacation!

**Comment by mstenseth** [comment]
in 2024 , i can make right click work..

**Comment by jettorio** [comment]
Same issue here. KDE Plasma 6.0.3-2, Arch, stable steam 1709846872
Was not happening couple updates back

**Comment by red15** [comment]
> **Edit:** this is seemingly 'fixed' for me by launching steam with `STEAM_DISABLE_BROWSER_SHUTDOWN_WORKAROUND=1 steam` (from here: [#10758 (comment)](https://github.com/ValveSoftware/steam-for-linux/issues/10758#issuecomment-2037102437)). Though now I'm getting steamwebhelper restarts with steam ui relaunching entirely, so not sure about the viability of this approach. I assume that the root causes of these issues are possibly connected.

For me this workaround means that when having the context menu open for a friend and hovering over a submenu it closes the entire steam UI and the terminal I launched it in spews out lines to show how the whole steamwebhelper.sh got restarted:

```
steamwebhelper.sh[53865]: === Sat Apr 13 10:33:08 AM CEST 2024 ===
steamwebhelper.sh[53865]: Starting steamwebhelper under bootstrap sniper steam runtime at /home/red15/.local/share/Steam/ubuntu12_64/steam-runtime-sniper
```

**Comment by RoadTrain** [comment]
@red15 Yep, I also get these UI restarts, though mainly when closing child windows (like Properties window of a game). Anyway, it's not a viable workaround, so I personally reverted back.

But the flickering was gone, so I figured this information might help eventually fixing the issue.

**Comment by corigne** [comment]
Any forward action on this? I'm also having this issue on i3 / arch.

**Comment by Zorrototo** [comment]
It's a graphical problem, there is some kind of video crash internally in Steam when it happens, unrelated to window rules or what you suggest. It is not a focus or similar issue.

**Comment by djhnsn** [comment]
Same problem for a long time. 

KDE Plasma 6
X11
Nvidia
Manjaro
Linux 6.6.30

**Comment by 576125176251825152** [comment]
Same issue here, I have had this issue since steam overhauled the friends list UI AFAIR.

DWM
X11
Nvidia
Arch Linux 
6.10.0-zen1-2-zen

**Comment by anthraxx** [comment]
For anyone who uses XMonad and still experiences this issue, please try the following:

1. Make sure you are using `xmonad-contrib` at least version `0.18.1` (not any version before it)
2. Make sure to use `XMonad.Hooks.EwmhDesktops` and use minimum `ewmh`:
```haskell
import XMonad
import XMonad.Hooks.EwmhDesktops

main = xmonad $ … . ewmh . … $ def {…}
```

Optional: Make sure to use `fixSteamFlicker` from `XMonad.Util.Hacks` if you experience flickering:
```haskell
import XMonad.Util.Hacks (fixSteamFlicker)

main = xmonad $ … . ewmhFullscreen . ewmh . … $ def {
        …
        handleEventHook = fixSteamFlicker <+> fullscreenEventHook,
}
```

**Comment by FossPrime** [comment]
I've been having the same issue for over a year. Game properties button doesn't work 20% of the time and Context menus in the Library tree appears and disappears within 100ms after **right click**, about 20% of the time. 

Changing the focus settings on Gnome Tweaks doesn't fix it.

Steam Beta Branch: Steam Beta Update
Steam Version: 1728093750
Steam Client Build Date: Fri, Oct 4 8:50 PM UTC -08:00
Steam Web Build Date: Fri, Oct 4 6:51 PM UTC -08:00
Steam API Version: SteamClient021
NixOS with Gnome on Linux 6.10.0

**Comment by gepbird** [comment]
@FossPrime try https://github.com/ValveSoftware/steam-for-linux/issues/9684#issuecomment-1683211268.

This fixed for me on NixOS using dwm.

**Comment by Zorrototo** [comment]
On my side since I switched from GTX 1060 to an RX 6800, I don't seem to have the problem anymore, I see for a split second a black menu in the nested menus, but it doesn't close anymore on itself immediately. I'm  sure NVIDIA was a big part of the problem on my side but it seems recent Steam also help "fix" (or made it better at least) the issue.

How many people with AMD card still have the issue? and if so maybe provide more system info.

**Comment by FossPrime** [comment]
> @FossPrime try [#9684 (comment)](https://github.com/ValveSoftware/steam-for-linux/issues/9684#issuecomment-1683211268).
> 
> This fixed for me on NixOS using dwm.

That didn't help at all for me on Gnome. I checked the Gnome issue tracker and there doesn't appear to be any discussion about this conflict. I don't think it's GPU related as I'm running mesa drivers on a 12-gen Intel 1545P, as stable as you can get. Changing resolution from 4k to 720p didn't help. I also have a 1540P laptop with ChromeOS, and the context menu work 100% of the time there, even though it's basically the same processor and it's running Wayland/Flatpak Steam.

I can't use Steam on my Chromebook as it doesn't support CJK fonts for game reviews/titles or browse files for patches and mods.

**Comment by Joshua079** [comment]
This really needs to be fixed. I can't seem to figure out why every other distros I've used, don't have this issue. Yet on Lubuntu, Steam's context menus are just broken. 
Settings > interface > "enable context menu focus compatibility mode" helps a little, but as others mentioned, it doesn't work for anything in the immediate context menu. Its enough for most features, but isn't a complete enough fix.

**Comment by JuniorWolfgamingDE** [comment]
I still experience this issue in the friend menu to this day. Is there no fix in sight?

**Comment by Raxyz0** [comment]
Also experiencing this in Steam Overlay whenever I try to invite friends. It's infuriating.

OS: CachyOS x86_64
Kernel: Linux 7.1.3-1-cachyos
DE: KDE Plasma 6.7.2
WM: KWin (Wayland)

**Comment by AudioEmu** [comment]
I can't add non-steam games to my library, and changing any settings requires switching to big picture mode and using a controller to navigate since all context menus seem to exist for a single frame before disappearing again.

OS: Arch x3_64
Kernel: Linux 7.2.6-arch2-1
DE: Niri
WM: Wayland

**Comment by kisak-valve** [comment]
@AudioEmu you're looking for #13566 instead of this issue report.
